sklearn/feature_selection/tests/test_sequential.py

```diff
@@ -23,39 +24,131 @@ def test_bad_n_features_to_select(n_features_to_select):
 
 def test_bad_direction():
     X, y = make_regression(n_features=5)
-    sfs = SequentialFeatureSelector(LinearRegression(), direction="bad")
+    sfs = SequentialFeatureSelector(
+        LinearRegression(), n_features_to_select="auto", direction="bad"
+    )
     with pytest.raises(ValueError, match="must be either 'forward' or"):
         sfs.fit(X, y)
 
 
 @pytest.mark.parametrize("direction", ("forward", "backward"))
-@pytest.mark.parametrize("n_features_to_select", (1, 5, 9, None))
+@pytest.mark.parametrize("n_features_to_select", (1, 5, 9, "auto"))
```

Comment:
Looking at this again, we need to continue to test `n_features_to_select=None` to make sure it works.

We can use a `@pytest.mark.filterwarnings("ignore:Leaving `n_features_to_select` to ")` to ignore the warning.

Reply:
add `None` with https://github.com/scikit-learn/scikit-learn/pull/20145/commits/0dd1499a646bcfa47fadf2d3ca2db9e8329e54ee